TypeScript compilation error of attributeChangedCallback for Polymer.Element

<!--
If you are asking a question rather than filing a bug, try one of these instead:
- StackOverflow (https://stackoverflow.com/questions/tagged/polymer)
- Polymer Slack Channel (https://bit.ly/polymerslack)
- Mailing List (https://groups.google.com/forum/#!forum/polymer-dev)
-->
<!-- Instructions For Filing a Bug: https://github.com/Polymer/polymer/blob/master/CONTRIBUTING.md#filing-bugs -->
### Description
<!-- Example: Error thrown when calling `appendChild` on Polymer element -->

I created my Polymer element as following with TypeScript:

```typescript
/// <reference path="./bower_components/polymer/types/polymer-element.d.ts"/>

class MyElement extends Polymer.Element {
    static get is() {
        return 'my-element';
    }

    static get properties() {
        return {};
    }
}
```

And I added a callback method to be notified the attribute changes:

```typescript
attributeChangedCallback(name: string, oldVal: string | null, newVal: string | null) {
    super.attributeChangedCallback(name, oldVal, newVal);
}
```

But `tsc` compiler raised an error as following:

```
index.ts:13:15 - error TS2339: Property 'attributeChangedCallback' does not exist on type 'Element'.
```

If my understanding about the Polymer elements' lifecycle is correct, a callback method in `super` instance must be called to handle lifecycle hooks properly. But `tsc` raises an error for it.

https://www.polymer-project.org/2.0/docs/devguide/custom-elements


#### Live Demo

https://github.com/rhysd/typescript-polymer-bug

#### Steps to Reproduce
<!--
Example:

1. Create `my-element`
2. Append `my-element` to document.body
3. Create `div`.
4. Append `div` to `my-element`
-->

1. `$ git clone https://github.com/rhysd/typescript-polymer-bug.git`
2. `$ cd typescript-polymer-bug`
3. `$ npm start`


#### Expected Results

Compilation is successfully done

#### Actual Results

Compilation error:

```
index.ts:13:15 - error TS2339: Property 'attributeChangedCallback' does not exist on type 'Element'.

13         super.attributeChangedCallback(name, oldVal, newVal);
                 ~~~~~~~~~~~~~~~~~~~~~~~~


```


### Browsers Affected
<!-- Check all that apply -->
- [x] Chrome
- [x] Firefox
- [x] Edge
- [x] Safari 9
- [x] Safari 8
- [x] IE 11

### Versions
<!--
`Polymer.version` will show the version for Polymer
`bower ls` or `npm ls` will show the version of webcomponents.js or webcomponents-lite.js
-->
- Polymer: v2.4.0
- webcomponents: v1.1.0
- TypeScript 2.7

--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
   },
   "devDependencies": {
     "@polymer/gen-closure-declarations": "^0.4.0",
-    "@polymer/gen-typescript-declarations": "^1.1.1",
+    "@polymer/gen-typescript-declarations": "^1.1.3",
     "@webcomponents/shadycss": "^1.1.0",
     "@webcomponents/webcomponentsjs": "^1.1.0",
     "babel-preset-minify": "^0.2.0",

--- a/types/lib/elements/array-selector.d.ts
+++ b/types/lib/elements/array-selector.d.ts
@@ -29,7 +29,7 @@ declare namespace Polymer {
    * representing the last selected item.  When `multi` is true, `selected`
    * is an array of multiply selected items.
    */
-  function ArraySelectorMixin<T extends new (...args: any[]) => {}>(base: T): T & ArraySelectorMixinConstructor & Polymer.ElementMixinConstructor;
+  function ArraySelectorMixin<T extends new (...args: any[]) => {}>(base: T): T & ArraySelectorMixinConstructor & Polymer.ElementMixinConstructor & Polymer.PropertyEffectsConstructor & Polymer.TemplateStampConstructor & Polymer.PropertyAccessorsConstructor & Polymer.PropertiesChangedConstructor & Polymer.PropertiesMixinConstructor;
 
   interface ArraySelectorMixinConstructor {
     new(...args: any[]): ArraySelectorMixin;

--- a/types/lib/legacy/legacy-element-mixin.d.ts
+++ b/types/lib/legacy/legacy-element-mixin.d.ts
@@ -26,7 +26,7 @@ declare namespace Polymer {
    * found on the Polymer 1.x `Polymer.Base` prototype applied to all elements
    * defined using the `Polymer({...})` function.
    */
-  function LegacyElementMixin<T extends new (...args: any[]) => {}>(base: T): T & LegacyElementMixinConstructor & Polymer.ElementMixinConstructor & Polymer.GestureEventListenersConstructor;
+  function LegacyElementMixin<T extends new (...args: any[]) => {}>(base: T): T & LegacyElementMixinConstructor & Polymer.ElementMixinConstructor & Polymer.PropertyEffectsConstructor & Polymer.TemplateStampConstructor & Polymer.PropertyAccessorsConstructor & Polymer.PropertiesChangedConstructor & Polymer.PropertiesMixinConstructor & Polymer.GestureEventListenersConstructor;
 
   interface LegacyElementMixinConstructor {
     new(...args: any[]): LegacyElementMixin;

--- a/types/lib/mixins/dir-mixin.d.ts
+++ b/types/lib/mixins/dir-mixin.d.ts
@@ -29,7 +29,7 @@ declare namespace Polymer {
    * - Changing `dir` at runtime is supported.
    * - Opting out of the global direction styling is permanent
    */
-  function DirMixin<T extends new (...args: any[]) => {}>(base: T): T & DirMixinConstructor & Polymer.PropertyAccessorsConstructor;
+  function DirMixin<T extends new (...args: any[]) => {}>(base: T): T & DirMixinConstructor & Polymer.PropertyAccessorsConstructor & Polymer.PropertiesChangedConstructor;
 
   interface DirMixinConstructor {
     new(...args: any[]): DirMixin;

--- a/types/lib/mixins/element-mixin.d.ts
+++ b/types/lib/mixins/element-mixin.d.ts
@@ -76,7 +76,7 @@ declare namespace Polymer {
    *   `observedAttributes` implementation will automatically return an array
    *   of dash-cased attributes based on `properties`)
    */
-  function ElementMixin<T extends new (...args: any[]) => {}>(base: T): T & ElementMixinConstructor & Polymer.PropertyEffectsConstructor & Polymer.PropertiesMixinConstructor;
+  function ElementMixin<T extends new (...args: any[]) => {}>(base: T): T & ElementMixinConstructor & Polymer.PropertyEffectsConstructor & Polymer.TemplateStampConstructor & Polymer.PropertyAccessorsConstructor & Polymer.PropertiesChangedConstructor & Polymer.PropertiesMixinConstructor;
 
   interface ElementMixinConstructor {
     new(...args: any[]): ElementMixin;

--- a/types/lib/mixins/property-effects.d.ts
+++ b/types/lib/mixins/property-effects.d.ts
@@ -46,7 +46,7 @@ declare namespace Polymer {
    * whereas the default when using `PropertyAccessors` standalone is to be
    * async by default.
    */
-  function PropertyEffects<T extends new (...args: any[]) => {}>(base: T): T & PropertyEffectsConstructor & Polymer.TemplateStampConstructor & Polymer.PropertyAccessorsConstructor;
+  function PropertyEffects<T extends new (...args: any[]) => {}>(base: T): T & PropertyEffectsConstructor & Polymer.TemplateStampConstructor & Polymer.PropertyAccessorsConstructor & Polymer.PropertiesChangedConstructor;
 
   interface PropertyEffectsConstructor {
     new(...args: any[]): PropertyEffects;

